src/macros.rs

```diff
@@ -714,55 +726,83 @@ macro_rules! clap_app {
+        }
+    };
+    (@app ($builder:expr) (@subcommand $name:ident => $($tail:tt)*) $($tt:tt)*) => {
+        $crate::clap_app!{ @app
+            ($builder.subcommand(
+                $crate::clap_app!{ @app ($crate::App::new(stringify!($name))) $($tail)* }
             ))
             $($tt)*
         }
     };
-// Yaml like function calls - used for setting various meta directly against the app
+    // Yaml like function calls - used for setting various meta directly against the app
     (@app ($builder:expr) ($ident:ident: $($v:expr),*) $($tt:tt)*) => {
-// clap_app!{ @app ($builder.$ident($($v),*)) $($tt)* }
         $crate::clap_app!{ @app
             ($builder.$ident($($v),*))
             $($tt)*
         }
     };
-
-// Add members to group and continue argument handling with the parent builder
+    // Add members to group and continue argument handling with the parent builder
     (@group ($builder:expr, $group:expr)) => { $builder.group($group) };
     // Treat the group builder as an argument to set its attributes
     (@group ($builder:expr, $group:expr) (@attributes $($attr:tt)*) $($tt:tt)*) => {
         $crate::clap_app!{ @group ($builder, $crate::clap_app!{ @arg ($group) (-) $($attr)* }) $($tt)* }
     };
+    (@group ($builder:expr, $group:expr) (@arg ($name:expr): $($tail:tt)*) $($tt:tt)*) => {
+        $crate::clap_app!{ @group
+            ($crate::clap_app!{ @app ($builder) (@arg ($name): $($tail)*) },
+             $group.arg($name))
+            $($tt)*
+        }
+    };
+    (@group ($builder:expr, $group:expr) (@arg $name:literal: $($tail:tt)*) $($tt:tt)*) => {
+        $crate::clap_app!{ @group
+            ($crate::clap_app!{ @app ($builder) (@arg $name: $($tail)*) },
+             $group.arg(stringify!($name).trim_matches('"')))
+            $($tt)*
+        }
+    };
     (@group ($builder:expr, $group:expr) (@arg $name:ident: $($tail:tt)*) $($tt:tt)*) => {
         $crate::clap_app!{ @group
             ($crate::clap_app!{ @app ($builder) (@arg $name: $($tail)*) },
              $group.arg(stringify!($name)))
             $($tt)*
         }
     };
-
-// No more tokens to munch
+    // No more tokens to munch
     (@arg ($arg:expr) $modes:tt) => { $arg };
-// Shorthand tokens influenced by the usage_string
+    // Shorthand tokens influenced by the usage_string
     (@arg ($arg:expr) $modes:tt --($long:expr) $($tail:tt)*) => {
         $crate::clap_app!{ @arg ($arg.long($long)) $modes $($tail)* }
     };
+    (@arg ($arg:expr) $modes:tt --$long:literal $($tail:tt)*) => {
+        $crate::clap_app!{ @arg ($arg.long(stringify!($long).trim_matches('"'))) $modes $($tail)* }
+    };
     (@arg ($arg:expr) $modes:tt --$long:ident $($tail:tt)*) => {
         $crate::clap_app!{ @arg ($arg.long(stringify!($long))) $modes $($tail)* }
     };
+    (@arg ($arg:expr) $modes:tt -($short:expr) $($tail:tt)*) => {
+        $crate::clap_app!{ @arg ($arg.short($short.chars().next().unwrap())) $modes $($tail)* }
+    };
```

Comment:
Hm, maybe just require a char instance/literal from users instead of .chars().next()?
```suggestion
    (@arg ($arg:expr) $modes:tt -($short:expr) $($tail:tt)*) => {
        $crate::clap_app!{ @arg ($arg.short($short)) $modes $($tail)* }
    };
```

Reply:
I think we should still have this because they might use multi character string here. It is also consistent with the derive. https://github.com/clap-rs/clap/blob/master/clap_derive/src/derives/attrs.rs#L144

Reply:
Ouch, this is the legacy of `structopt` - `Arg::short` takes `&str` in clap 2.x, but it's `char` in 3.x.

Users *used to* be able to use multi character strings here, but they aren't anymore.

We also need to fix tests/examples/docs/implementation in derive.